firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -401,13 +406,14 @@ private InstallationResponse readCreateResponse(HttpURLConnection conn) throws I
     }
     reader.endObject();
     reader.close();
-
+    inputStream.close();
     return builder.setResponseCode(ResponseCode.OK).build();
   }
 
   // Read the response from the generateAuthToken FirebaseInstallation API.
   private TokenResult readGenerateAuthTokenResponse(HttpURLConnection conn) throws IOException {
-    JsonReader reader = new JsonReader(new InputStreamReader(conn.getInputStream(), UTF_8));
+    InputStream inputStream = conn.getInputStream();
+    JsonReader reader = new JsonReader(new InputStreamReader(inputStream, UTF_8));
```

Comment:
in a future CL we should probably remove this code duplication

Reply:
Ack.

Reply:
Saw this again. Happy to chat offline.

Reply:
We can discuss offline and if there are changes, they will be part of another PR.